Samples/CameraSettingsExternalSettingsApp/DefaultControlHelper/DefaultControlHelper.cpp

```diff
@@ -46,190 +46,258 @@ namespace winrt::DefaultControlHelper::implementation
+                ULONG cbControl = 0;
+                void* pData = nullptr;
+                ULONG cbData = 0;
+
+                THROW_IF_FAILED(m_controlDefaults->LockControlData(&pControl, &cbControl, &pData, &cbData));
                 {
-                    if (m_rangeInfo.has_value())
+                    THROW_IF_NULL_ALLOC(pControl);
+                    THROW_IF_NULL_ALLOC(pData);
+                    auto unlock = wil::scope_exit([&]()
+                        {
+                            // The Unlock call can fail if there's no existing configuration or the required flags/capabilities
+                            // are otherwise wrong or unset.
+                            THROW_IF_FAILED(m_controlDefaults->UnlockControlData());
+                        });
+
+                    resultValue = m_internalController->GetStoredDefaultValue((void*)pControl, cbControl, (void*)pData, cbData);
+
+                    // Getting range info if available
+                    MF_CAMERA_CONTROL_RANGE_INFO rangeInfo = {};
+
+                    try
+                    {
+                        THROW_IF_FAILED(m_controlDefaults->GetRangeInfo(&rangeInfo));
+                        m_rangeInfo.emplace(rangeInfo);
+                    }
+                    catch (const wil::ResultException& e)
                     {
-                        valueOut = m_rangeInfo->defaultValue;
+                        // GetRangeInfo will return MF_E_NOT_FOUND if the control does
+                        // not support range information.
+                        if (e.GetErrorCode() != MF_E_NOT_FOUND)
+                        {
+                            THROW_EXCEPTION(e);
+                        }
                     }
                 }
-            });
-
-            valueOut = pControl->Value;
+            }
+            catch (...)
+            {
+                if (m_rangeInfo.has_value())
+                {
+                    resultValue = m_rangeInfo->defaultValue;
+                }
+            }
         }
-
-        return m_defaultValue = valueOut;
+        
+        return resultValue;
     }
 
-    void DefaultController::DefaultValue(uint32_t const& value)
+    void DefaultController::DefaultValue(int32_t const& value)
     {
-        m_internalController->SaveDefault(m_controlDefaults, value);
+        void* pControl = nullptr;
+        ULONG cbControl = 0;
+        void* pData = nullptr;
+        ULONG cbData = 0;
 
-        auto strongManager = m_controlManager.get();
-        THROW_HR_IF_NULL(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), strongManager);
-        
+        auto collection = m_controlManager->GetCollection();
 
-        THROW_IF_FAILED(strongManager->Save());
-        m_defaultValue = value;
+        m_internalController->Initialize(m_controlManager, m_controlDefaults, m_setGuid, m_id);
+
+        THROW_IF_FAILED(m_controlDefaults->LockControlData(&pControl, &cbControl, &pData, &cbData));

```

Comment:
>LockControlData [](http://example.com/codeflow?start=43&length=15)

not really following why you moved lock/unlock here from SaveDefault?

Reply:
this way I write it only once but then call the internalController only to interpret the payload